pkg/build/builder/common.go

```diff
@@ -201,15 +217,24 @@ func retryBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface,
 		latestBuild.Status.Output.To = build.Status.Output.To
 		latestBuild.Status.Stages = build.Status.Stages
 
-		if _, err := client.UpdateDetails(latestBuild); err != nil {
-			return err
+		_, err = client.UpdateDetails(latestBuild)
+		netErr, ok := err.(net.Error)
+
+		switch {
+		case err == nil:
+			return true, nil
+		case errors.IsConflict(err):
+			latestBuild = nil
+			return false, err
+		case ok && netErr.Temporary():
+			return false, err
```

Comment:
if you return an error from within your function, ExponentialBackoff will stop retrying.  I don't think that's what you want here.  You need to return nil so it will retry your function.